link-area-delegation-explainer.md

```diff
@@ -0,0 +1,141 @@
+
+We could have different similar alternatives to this with different ergonomic tradeoffs, e.g. using ID references:
+```html
+<section class="card" link="card123-title">
+  <a href="/post?id=123" id="card123-title">Post Title</a>
+  <img>
+  <button>Join</button>
+  <button>Share</button>
+</section>
+```
+
+Note that in this example, the link still has to be a descendant of the card. This is to guide web authors to an interactive structure where the links are in the appropriate place in the tree.
+
+## Alternatives considered
+
+### A new element type
+One of the alternatives that sound natural is a new element type:
+```html
+<section class="card">
+  <linkarea>
+    <a href="/post?id=123" defaultlink>Post Title</a>
+    <img>
+    <button>Join</button>
+    <button>Share</button>
+  </linkarea>
+</section>
+```
+
+This can feel compelling in a semantic sense, but becomes undesirable when we think of the flexibility of attributes.
+For example, a use case for link area delegation is making table rows clickable.
+
+```html
+<table>
+  <tr linkarea>
+     <td>1.</td>
+     <td><a href="/details?id=1" defaultlink>Title</a>
+  </tr>
+</table>
+```
+
+This is not possible with a new element, as `tr` (and some other container types) have special HTML parser behavior, and allowing them to include new types of elements would
+break the page in browsers that don't support the new element type.
+
+### CSS instead of HTML
+
+Since this proposal focuses on click areas, perhaps CSS is better suited to this than HTML?
+Something like:
+
+```html
+<section class="card">
+  <a href="/post?id=123">Post Title</a>
+  <img>
+  <button>Join</button>
+  <button>Share</button>
+</section>
+```
+
+```css
+.card { pointer-area: contain; }
+.card a:nth-child(1) { pointer-area: expand; }
+```
+
+This has nice ergonomics because the selection of link area can be done outside of the main structure of the page using CSS selectors.
+However, something about this feels limiting, as it is constrained to click areas in advance, and doesn't lend itself to future UA enhancements of this experience, such as context menus.
+
+### Use invokers
+One interesting alternative from @jaffathecake is to use [invokers](https://open-ui.org/components/invokers.explainer/) for this, e.g.:
+```html
+<section class="card" commandfor="card123-title" command="click">
+  <a href="/post?id=123" id="card123-title">Post Title</a>
+  <img>
+  <button>Join</button>
+  <button>Share</button>
+</section>
+```
+
+This has the nice effect of being more general purpose, e.g. delegating any click rather than just links, but perhaps the special-purpose link delegation has some other added value.
+
+## Open questions
+- Do link area ancestors have a default UA styles (probably not?)
+- Can you nest link areas? (probably?)
```

Comment:
You should be able nest link area but there should not be an intermediate linkarea between a link area and it's target link.

OK:
```html
<section class="card" commandfor="card123-title" commands="click contextmenu">
  <a href="/post?id=123" id="card123-title">Post Title</a>
  <img>
  <div class="card" commandfor="card123-img" commands="click contextmenu">
    <a href="/post?id=123" id="card123-img">
      <img>
    </a>
    <button>edit</button>
  </div>

  <button>Join</button>
  <button>Share</button>
</section>
```


NOT OK:
```html
<section class="card" commandfor="card123-title" commands="click contextmenu">  
  <img>
  <div class="card" commandfor="card123-img" commands="click contextmenu">
    <a href="/post?id=123" id="card123-title">Post Title</a>
    <a href="/post?id=123" id="card123-img">
      <img>
    </a>
    <button>edit</button>
  </div>
  <button>Join</button>
  <button>Share</button>
</section>
```

Reply:
Let's add this for a discussion later rather than in the PR for the initial explainer?

Reply:
Let's add this to the discussion once the PR is merged

Reply:
it might be good to fill out those examples a bit more, or potentially provide a visual as well.  looking at just the markup, i'm having a hard time seeing these as use cases that'd necessitate this feature.